api/src/shared/application/scripts/parsers.js

```diff
@@ -27,6 +29,68 @@ export function csvFileParser(columnSchema = []) {
   };
 }
 
+/**
+ * Stream CSV file by chunk with the given column schema
+ * @param {Record<string, Joi.Schema>} columnSchema
+ * @returns {Promise<Function>} function to process each chunk
+ */
+export function csvFileStreamer(columnSchema = []) {
+  return async (filePath) => {
+    return async (onChunk, chunkSize = 1) => {
+      const dataStream = fs.createReadStream(filePath);
+      const parseStream = await streamCsv({ header: true, skipEmptyLines: true });
+      dataStream.pipe(parseStream);
+
+      let chunk = [];
+      let processing = Promise.resolve();
+
+      return new Promise((resolve, reject) => {
+        parseStream.on('data', (row) => {
+          try {
+            const validatedRow = Joi.attempt(
+              row,
+              Joi.object().keys(
+                columnSchema.reduce((acc, column) => {
+                  acc[column.name] = column.schema;
+                  return acc;
+                }, {}),
+              ),
+              row,
+            );
+
+            chunk.push(validatedRow);
+
+            if (chunk.length >= chunkSize) {
+              const currentChunk = [...chunk];
+              chunk = [];
+              processing = processing.then(() => onChunk(currentChunk));
```

Comment:
Petite suggestion pour, je pense, rendre l'intention plus claire : 
```suggestion
              const chunkCopy = [...chunk];
              chunk = [];
              processing = processing.then(() => onChunk(chunkCopy));
```